Shared/mods/deathmatch/logic/luadefs/CLuaUtilDefs.cpp

```diff
@@ -716,3 +719,40 @@ int CLuaUtilDefs::tocolor(lua_State* luaVM)
     lua_pushnumber(luaVM, static_cast<lua_Number>(ulColor));
     return 1;
 }
+
+int CLuaUtilDefs::Require(lua_State* luaVM)
+{
+    //  table require ( string name )
+    SString strMod;
+
+    CLuaMain*        pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadString(strMod);
+
+    if (!argStream.HasErrors())
+    {
+        SString strError = "";
+        // Check if package exists already, if so load it
+        // stack: ["moduleName"]
+        pLuaMain->GetPackage(luaVM, strMod);            // stack: ["moduleName",pkgModule/nil]
+        if (lua_type(luaVM, -1) != LUA_TNIL)
+            return 1;
+        lua_pop(luaVM, 1);            // stack: ["moduleName"]
+
+        // Check whether the appropriate pure lua module exists
+        if (pLuaMain->LoadLuaLib(luaVM, strMod, strError))            // stack: ["moduleName",pkgLuaMod/nil]
+            return 1;
+#ifndef MTA_CLIENT
+        strError += "\n";
+        // Check if a C library exists
```

Comment:
libraries in c/c++ will allowed?

Reply:
Yes. See commit message of https://github.com/multitheftauto/mtasa-blue/commit/a59316354482ee89687a95d238ee0078ae006a6d

Put behind ACL too.